src/app/components/booking/index.tsx

```diff
@@ -166,17 +171,31 @@ export default function BookingComponent({
       return setOpenPrepaymentModal(true);
     }
 
+    if (totalPayment > 0) {
+      getCheckoutId()
+      return
+    }
```

Comment:
This if needs to be inside of updateNotion. 

Otherwise if they open the popup and they need to pay it will skip the payment.